runtime/common/metering.go

```diff
@@ -145,10 +153,17 @@ func NewDictionaryMemoryUsages(length int) (MemoryUsage, MemoryUsage) {
 		}
 }
 
-func NewDictionarySizeUsage(length int) MemoryUsage {
+func NewDictionaryAdditionalSizeUsage(originalSize, additionalSize int) MemoryUsage {
+	var newAmount uint64
+	if originalSize <= 1 {
+		newAmount = uint64(originalSize + additionalSize)
+	} else {
+		// size of b+ tree grows logarithmically with the size of the tree
+		newAmount = uint64(math.Log2(float64(originalSize)) + float64(additionalSize))
```

Comment:
Followed up on the formula here, not quite sure log2 is a correct over-estimation for the atree data structures: 
https://axiomzen.slack.com/archives/C0355LG8V6J/p1649889273942269?thread_ts=1649706053.685719&cid=C0355LG8V6J